test/win-bats/collection.xml

```diff
@@ -1724,7 +1724,7 @@
     call :verify_line -1 x "*** Error running the test suite"
 	</case>
 
-	<case name="Non-boolean @test (multiple xs:boolean) with no comparison factors (XQuery)">
+	<case ifdef="SAXON12_INITIAL_ISSUES_FIXED" name="Non-boolean @test (multiple xs:boolean) with no comparison factors (XQuery)">
```

Comment:
@AirQuick : I reproduced the difference in behavior between Saxon 11 and Saxon 12 with the small query below, but I'm not sure if it's a Saxon 12 bug or a legitimate change in behavior.
```
xquery version "3.1";
let $test-result as item()* := (true(), true())
return
if ($test-result instance of xs:boolean)
then 'Saxon 12 errors out even if we do not get here at run time. ' || boolean($test-result)
else 'Got to else'
```
Outputs:
```
java -cp "C:\Program Files\Saxonica\SaxonHE11.5J\saxon-he-11.5.jar" net.sf.saxon.Query -q:repro-issue-multiple-boolean.xq
<?xml version="1.0" encoding="UTF-8"?>Got to else

java -cp "C:\Program Files\Saxonica\SaxonHE12.0J\saxon-he-12.0.jar" net.sf.saxon.Query -q:repro-issue-multiple-boolean.xq
<?xml version="1.0" encoding="UTF-8"?>net.sf.saxon.trans.UncheckedXPathException: Effective boolean value is not defined for a sequence of two or more items starting with a boolean
        at net.sf.saxon.expr.Literal$LiteralElaborator.elaborateForBoolean(Literal.java:907)
        at net.sf.saxon.functions.BooleanFn$BooleanFnElaborator.elaborateForBoolean(BooleanFn.java:173)
        at net.sf.saxon.expr.elab.BooleanElaborator.elaborateForPull(BooleanElaborator.java:24)
        at net.sf.saxon.expr.elab.Elaborator.lazily(Elaborator.java:110)
        at net.sf.saxon.expr.FunctionCall$FunctionCallElaborator.allocateArgumentEvaluators(FunctionCall.java:633)
        at net.sf.saxon.expr.FunctionCall$FunctionCallElaborator.setExpression(FunctionCall.java:618)
        at net.sf.saxon.expr.SystemFunctionCall$SystemFunctionCallElaborator.setExpression(SystemFunctionCall.java:587)
        at net.sf.saxon.expr.Expression.makeElaborator(Expression.java:1768)
        at net.sf.saxon.expr.instruct.Choose$ChooseExprElaborator.elaborateForPush(Choose.java:1165)
        at net.sf.saxon.query.XQueryExpression.processQuery(XQueryExpression.java:495)
        at net.sf.saxon.query.XQueryExpression.run(XQueryExpression.java:473)
        at net.sf.saxon.s9api.XQueryEvaluator.run(XQueryEvaluator.java:464)
        at net.sf.saxon.Query.runQuery(Query.java:975)
        at net.sf.saxon.Query.doQuery(Query.java:444)
        at net.sf.saxon.Query.main(Query.java:106)
Caused by: net.sf.saxon.trans.XPathException: Effective boolean value is not defined for a sequence of two or more items starting with a boolean
        at net.sf.saxon.expr.parser.ExpressionTool.ebvError(ExpressionTool.java:767)
        at net.sf.saxon.expr.parser.ExpressionTool.effectiveBooleanValue(ExpressionTool.java:677)
        at net.sf.saxon.value.SequenceExtent$Of.effectiveBooleanValue(SequenceExtent.java:184)
        at net.sf.saxon.expr.Literal$LiteralElaborator.elaborateForBoolean(Literal.java:904)
        ... 14 more
Fatal error during query: net.sf.saxon.trans.UncheckedXPathException: Effective boolean value is not defined for a sequence of two or more items starting with a boolean
```

 If it's not a Saxon 12 bug or if we don't want to wait for a fix, one idea for working around it is to replace `boolean($local:test-result)` with `boolean($local:test-result[1])` in the [XQuery compile_expect.xsl file](https://github.com/xspec/xspec/blob/673d5d75a35cefdfc69d88208882f255fd227ef2/src/compiler/xquery/compile/compile-expect.xsl#LL74C32-L74C59).

FYI, BaseX 10.6 can execute the small query without an error. I noticed from the compiler log that BaseX recognizes the data type mismatch but later recognizes that it won't reach the mismatch at run time (`rewrite if: if(false()) { error(Q{http://www.w3.org/2005/xqt-errors}FORG0006, "Single xs:boolean expected, sequence found: (true(), true()).") } else { "Got to else" } -> "Got to else"`).

Reply:
Logged as https://saxonica.plan.io/issues/6082